src/main/java/net/spy/memcached/MemcachedConnection.java

```diff
@@ -587,26 +587,24 @@ private Set<ArcusReplNodeAddress> getSlaveAddrsFromGroupAddrs(
   /* ENABLE_REPLICATION end */
 
   /* ENABLE_REPLICATION if */
-  private void switchoverMemcachedReplGroup(MemcachedNode node, boolean cancelNonIdempotent) {
-    MemcachedReplicaGroup group = node.getReplicaGroup();
+  private void switchoverMemcachedReplGroup(MemcachedReplicaGroup group,
+                                            boolean cancelNonIdempotent) {
+
+    MemcachedNode oldMaster = group.getMasterNode();
 
     /*  must keep the following execution order when switchover
      * - first moveOperations
      * - second, queueReconnect
      *
      * because moves all operations
      */
-    if (group.getMasterNode() != null && group.getMasterCandidate() != null) {
-      if (((ArcusReplNodeAddress) node.getSocketAddress()).isMaster()) {
-        ((ArcusReplKetamaNodeLocator) locator).switchoverReplGroup(group);
-      }
-      node.moveOperations(group.getMasterNode(), cancelNonIdempotent);
-      addedQueue.offer(group.getMasterNode());
-      queueReconnect(node, ReconnDelay.IMMEDIATE,
-          "Discarded all pending reading state operation to move operations.");
-    } else {
-      getLogger().warn("Delay switchover because invalid group state : " + group);
-    }
+    ((ArcusReplKetamaNodeLocator) locator).switchoverReplGroup(group);
```

Comment:
기존의 `group.getMasterNode() != null && group.getMasterCandidate() != null` 조건이 만족할 때에만 switchover + move operation + old master reconnect 해야 하지 않나요?

Reply:
https://github.com/jam2in/arcus-works/issues/637#issuecomment-2521925920

해당 조건에 맞지 않는데도 이 메소드를 호출하는 경우가 로직적으로는 없어서 조건문을 제거했습니다.
그래도 조건문을 유지하도록 할까요?

Reply:
private 메서드라서 반드시 유지할 필요는 없어보이지만, 단순 검증 용도로 둔다면 나중에 리팩토링 시에 이 부분을 인지하여 버그 발생 가능성이 줄어들 것 같습니다.

Reply:
아래와 같이 조건을 넣어 두시죠.
- oldMaster는 분명히 존재하는 상황에서 switchover 대상이 될 수 있음.
- masterCandidate가 null인 경우는 없을 것으로 보이지만,
  masterCandidate가 null인 경우에 locator.switchoverReplGroup(group) 수행은 
  오동작(invalid group으로 변환)하게 되므로,
  masterCandidate != null 조건을 넣어두는 것이 괜찮을 것 같습니다.

```java
if (group.getMasterCandidate() != null) {
  ((ArcusReplKetamaNodeLocator) locator).switchoverReplGroup(group);
   . . .
   queueReconnect(oldMaster, ReconnDelay.IMMEDIATE,
        "Discarded all pending reading state operation to move operations.");
} else {
  getLogger().warn("Delay switchover because invalid group state : " + group);
}
```